### PR TITLE
ci: simplify cypress startup commands

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [x] Build or CI related changes
- [ ] Documentation content changes

## What is the new behavior?

```
$ nx e2e
```

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/demo/project.json [hunk 1]
```diff
@@ -275,6 +277,37 @@
...
+            "builder": "@nrwl/workspace:run-commands",
+            "options": {
+                "parallel": false,
+                "commands": [
+                    "npx wait-on http://localhost:3333 --timeout 1000",
+                    "cypress open --browser chrome --project ./projects/demo-integrations"
+                ]
+            }
+        },
+        "e2e:headless": {
```
Comment (nsbarsukov) [inline projects/demo/project.json:291]:
```suggestion
        "e2e-headless": {
```
(for uniformity of all commands' names)

File: projects/demo/project.json [hunk 2]
```diff
@@ -275,6 +277,37 @@
...
+            "builder": "@nrwl/workspace:run-commands",
+            "options": {
+                "parallel": false,
+                "commands": [
+                    "npx wait-on http://localhost:3333 --timeout 1000",
+                    "cypress open --browser chrome --headless --project ./projects/demo-integrations"
+                ]
+            }
+        },
+        "e2e:ci": {
```
Comment (nsbarsukov) [inline projects/demo/project.json:301]:
```suggestion
        "e2e-ci": {
```
(for uniformity of all commands' names)

**Review by nsbarsukov — APPROVED** [review]

File: .github/workflows/e2e.yml [hunk 3]
```diff
@@ -125,19 +120,14 @@ jobs:
...
       - name: Run screenshot tests on ${{ env.BASE_REF_DEMO }}
         continue-on-error: true
         env: # note: need to collect all
           # diff files for lumberjack bot
           CYPRESS_failOnSnapshotDiff: 0
-        run: |
-          npm run concurrently \
-              "npm run cy2:run:demo:next -- --spec '**/${{ matrix.specFolder }}/**/*.spec.ts'" \
-              "sleep 20 && npm run cy2:run:demo:next -- --spec '**/${{ matrix.specFolder }}/**/*.spec.ts'"
+        run: npx nx e2e:ci --key next --folder ${{ matrix.specFolder }}
```
Comment (nsbarsukov) [inline .github/workflows/e2e.yml:130]:
```suggestion
        run: npx nx e2e-ci --key next --folder ${{ matrix.specFolder }}
```

File: .github/workflows/e2e.yml [hunk 4]
```diff
@@ -94,15 +94,10 @@ jobs:
...
-          npx nx serve-compiled -- --path ${{ env.LOCAL_REF_DEMO }} \
-            & npx wait-on http://localhost:${{ env.NG_SERVER_PORT }}
+        run: npx nx serve-compiled --path ${{ env.LOCAL_REF_DEMO }} --port ${{ env.NG_SERVER_PORT }}
 
       - name: Run screenshot tests on ${{ env.LOCAL_REF_DEMO }}
-        run: |
-          npm run concurrently \
-              "npm run cy2:run:localhost -- --spec '**/${{ matrix.specFolder }}/**/*.spec.ts'" \
-              "sleep 20 && npm run cy2:run:localhost -- --spec '**/${{ matrix.specFolder }}/**/*.spec.ts'"
+        run: npx nx e2e:ci --key localhost --folder ${{ matrix.specFolder }}
```
Comment (nsbarsukov) [inline .github/workflows/e2e.yml:100]:
```suggestion
        run: npx nx e2e-ci --key localhost --folder ${{ matrix.specFolder }}
```

File: projects/demo/project.json (hunk 1 above)
Comment (splincode) [reply projects/demo/project.json]: fixed

File: projects/demo/project.json (hunk 2 above)
Comment (splincode) [reply projects/demo/project.json]: fixed

File: .github/workflows/e2e.yml (hunk 3 above)
Comment (splincode) [reply .github/workflows/e2e.yml]: fixed

File: .github/workflows/e2e.yml (hunk 4 above)
Comment (splincode) [reply .github/workflows/e2e.yml]: fixed

**Review by vladimirpotekhin — APPROVED** [review]

**Pull request merged**